tests/eas/preliminary.py

```diff
@@ -133,37 +133,55 @@ def _run_sysbench_work(self, cpu, duration):
 
     def _check_work_throughput(self, cpu):
         seconds = 1.0
-        margin = 0.2
+        margin = 0.1
         frequencies = self.target.cpufreq.list_frequencies(cpu)
         if len(frequencies) == 1:
             return True
+
         original_governor = self.target.cpufreq.get_governor(cpu)
         original_freq = None
         if original_governor == 'userspace':
             original_freq = self.target.cpufreq.get_frequency(cpu)
-        # set userspace governor
+
+        # Set userspace governor
         self.target.cpufreq.set_governor(cpu, 'userspace')
-        # do each freq in turn
+        # Do each freq in turn
         result = {}
         for freq in frequencies:
             self.target.cpufreq.set_frequency(cpu, freq)
             result[freq] = self._run_sysbench_work(cpu, seconds)
-        # restore governor
+
+        # Restore governor
         self.target.cpufreq.set_governor(cpu, original_governor)
         if original_freq:
             self.target.cpufreq.set_frequency(cpu, original_freq)
-        # compare work throughput
-        return result[frequencies[0]] < result[frequencies[-1]]
+
+        # Make sure work done at highest OPP is at least 10% higher
+        # than work done at lowest OPP - this filters the
+        # +/- 1 sysbench result noise
+        work_diff = result[frequencies[-1]] - result[frequencies[0]]
+        ok = work_diff > result[frequencies[0]] * margin
+        return (ok, result)
 
     def test_work_throughput(self):
-        failed_cpus = []
+        failed_results = {}
+
+        # Run test on each cpu
         for cpulist in self.env.topology.get_level('cpu'):
             cpu = cpulist[0]
-            if not self._check_work_throughput(cpu):
-                failed_cpus.append(cpu)
+            ok, result = self._check_work_throughput(cpu)
+            if not ok:
+                failed_results[cpu] = result
+
+        # Format error message
         msg='Work done did not scale with CPU Freq on CPUs: {}'\
-            .format(failed_cpus)
-        self.assertFalse(len(failed_cpus), msg=msg)
+            .format(failed_results.keys())
+        for cpu in failed_results.iterkeys():
+            msg += '\nCPU {}:'.format(cpu)
+            for freq, work in failed_results[cpu].iteritems():
+                msg += '\n\t{}Hz -> {} events'.format(freq, work)
```

Comment:
Maybe say '\n\t{}Hz -> processed {} events' to clarify that the "event" count is a measure of work throughput.